LedgerHQ/ledger-live-desktop
# LL-5232 USB troubleshooting wizard for LLD

## 🦒 Context (issues, jira)

> ~Leaving as draft until I've finished the spectron testing~ Now with spectron coverage
[LL-5232]
## 💻  Description / Demo (image or video)
<img width="1136" alt="image" src="https://user-images.githubusercontent.com/4631227/132360461-7beb139e-8afd-4e0a-ac87-8c0014b1c6a3.png">

This pr introduces a new wizard for troubleshooting the most common USB problems from our users, it's a pretty straightforward trial and error approach, with a continuous connection tested component to see if we've succeeded or not. If in the end we've not succeeded we can either trigger the Device repair modal for LNS or forward the case to support for other devices.

It can be accessed whenever there's a device action and we don't have a device plugged in, after 15 seconds there should be a new UI element that allows us to see this.

QA test plan notes:
- If we restart the app while it's showing a solution, the app should automatically show that solution on relaunch.
- Navigating away from the troubleshooter is the same as exiting it, meaning a restart will no longer bring us back there.
- A successful connection can be achieved by connecting a device an unlocking it in any of the solutions                                  
- There's an exposed .env to simulate windows/mac/linux `USBTROUBLESHOOTING_PLATFORM=mac`



## 🖤  Expectations to reach

PR must pass CI, merge develop if conflicts, do not force push. Thanks!

- **on QA**: at least one of these two checkboxes must be checked:
- **on delivery**: at least one of these two checkboxes must be checked: 

PR must pass CI, merge develop if conflicts, do not force push. Thanks!




[LL-5232]: https://ledgerhq.atlassian.net/browse/LL-5232
